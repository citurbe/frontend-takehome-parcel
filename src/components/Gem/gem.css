.gem-card {
  border: 3px solid grey;
  border-radius: 5px;
  margin: 10px;
  margin-left: 0;
  width: 48%;
  padding: 5px;
}

.save-button {
  margin-left: 65%;
  width: 35%;
  border-radius: 5px;
}
