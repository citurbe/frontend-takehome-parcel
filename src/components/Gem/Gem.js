import React from 'react';
import './gem.css';

const Gem = props => {
  return (
    <div className="gem-card">
      <a href={props.project_uri} target="blank">
        {props.name}
      </a>
      <p>version: {props.version}</p>
      <button
        className="save-button"
        onClick={() => props.toggleSave(props.name)}
      >
        {props.isSaved ? 'Un-Save' : 'Save'}
      </button>
    </div>
  );
};

export default Gem;
