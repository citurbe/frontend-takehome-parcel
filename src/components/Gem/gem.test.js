import React from 'react';
import Gem from './Gem';
import renderer from 'react-test-renderer';

const gemInfo = {
  name: 'Test Gem',
  version: '2.5',
  project_uri: 'http://fakeurl.com'
};

test('gem display renders correctly', () => {
  const component = renderer.create(<Gem isSaved={false} {...gemInfo} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
