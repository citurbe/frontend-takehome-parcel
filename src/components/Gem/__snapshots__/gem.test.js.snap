// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gem display renders correctly 1`] = `
<div
  className="gem-card"
>
  <a
    href="http://fakeurl.com"
    target="blank"
  >
    Test Gem
  </a>
  <p>
    version: 
    2.5
  </p>
  <button
    className="save-button"
    onClick={[Function]}
  >
    Save
  </button>
</div>
`;
